lte/gateway/c/core/oai/tasks/grpc_service/AmfServiceImpl.cpp

```diff
@@ -122,30 +129,125 @@ Status AmfServiceImpl::SetSmfSessionContext(
-  itti_msg.session_ambr.uplink_units = req_m5g.subscribed_qos().apn_ambr_ul();
-
-  itti_msg.session_ambr.downlink_unit_type = req_m5g.subscribed_qos().br_unit();
-  itti_msg.session_ambr.downlink_units = req_m5g.subscribed_qos().apn_ambr_dl();
-
-  // authorized qos profile
-  itti_msg.qos_list.qos_flow_req_item.qos_flow_identifier =
-      req_m5g.subscribed_qos().qos_class_id();
-
-  itti_msg.qos_list.qos_flow_req_item.qos_flow_level_qos_param
-      .qos_characteristic.non_dynamic_5QI_desc.fiveQI =
-      req_m5g.subscribed_qos().qos_class_id();  // enum
-  itti_msg.qos_list.qos_flow_req_item.qos_flow_level_qos_param
-      .alloc_reten_priority.priority_level =
-      req_m5g.subscribed_qos().priority_level();  // uint32
-  itti_msg.qos_list.qos_flow_req_item.qos_flow_level_qos_param
-      .alloc_reten_priority.pre_emption_cap =
-      (pre_emption_capability)req_m5g.subscribed_qos()
-          .preemption_capability();  // enum
-  itti_msg.qos_list.qos_flow_req_item.qos_flow_level_qos_param
-      .alloc_reten_priority.pre_emption_vul =
-      (pre_emption_vulnerability)req_m5g.subscribed_qos()
-          .preemption_vulnerability();  // enum
+  if (req_m5g.has_subscribed_qos()) {
+    itti_msg.session_ambr.uplink_unit_type = req_m5g.subscribed_qos().br_unit();
+    itti_msg.session_ambr.uplink_units = req_m5g.subscribed_qos().apn_ambr_ul();
+
+    itti_msg.session_ambr.downlink_unit_type =
+        req_m5g.subscribed_qos().br_unit();
+    itti_msg.session_ambr.downlink_units =
+        req_m5g.subscribed_qos().apn_ambr_dl();
+
+    // authorized qos profile
+    itti_msg.qos_flow_list.item[i].qos_flow_req_item.qos_flow_identifier =
+        req_m5g.subscribed_qos().qos_class_id();
+
+    itti_msg.qos_flow_list.item[i]
+        .qos_flow_req_item.qos_flow_level_qos_param.qos_characteristic
+        .non_dynamic_5QI_desc.fiveQI =
+        req_m5g.subscribed_qos().qos_class_id();  // enum
+    itti_msg.qos_flow_list.item[i]
+        .qos_flow_req_item.qos_flow_level_qos_param.alloc_reten_priority
+        .priority_level = req_m5g.subscribed_qos().priority_level();  // uint32
+    itti_msg.qos_flow_list.item[i]
+        .qos_flow_req_item.qos_flow_level_qos_param.alloc_reten_priority
+        .pre_emption_cap = (pre_emption_capability)req_m5g.subscribed_qos()
+                               .preemption_capability();  // enum
+    itti_msg.qos_flow_list.item[i]
+        .qos_flow_req_item.qos_flow_level_qos_param.alloc_reten_priority
+        .pre_emption_vul = (pre_emption_vulnerability)req_m5g.subscribed_qos()
+                               .preemption_vulnerability();  // enum
+    i++;
+  }
+
+  for (; i < req_m5g.qospolicy_size(); i++) {
+    ul_tft = &itti_msg.qos_flow_list.item[i].qos_flow_req_item.ul_tft;
+    memset(ul_tft, 0, sizeof(traffic_flow_template_t));
+    auto& qos_rule = req_m5g.qospolicy(i);
+    itti_msg.qos_flow_list.item[i].qos_flow_req_item.qos_flow_identifier =
```

Comment:
Follow same approach for "itti_msg.qos_flow_list.item[i].qos_flow_req_item" as in line number 164